client/document.go

```diff
@@ -201,25 +439,48 @@ func (doc *Document) GetValueWithField(f Field) (Value, error) {
 // JSON Merge Patch object. Note: fields indicated as nil in the Merge
 // Patch are to be deleted
 // @todo: Handle sub documents for SetWithJSON
-func (doc *Document) SetWithJSON(patch []byte) error {
-	var patchObj map[string]any
-	err := json.Unmarshal(patch, &patchObj)
+func (doc *Document) SetWithJSON(obj []byte, sd SchemaDescription) error {
+	v, err := fastjson.ParseBytes(obj)
 	if err != nil {
 		return err
 	}
+	o, err := v.Object()
+	if err != nil {
+		return err
+	}
+
+	return doc.setWithFastJSONObject(o, sd)
+}
 
-	for k, v := range patchObj {
-		err = doc.Set(k, v)
+func (doc *Document) setWithFastJSONObject(obj *fastjson.Object, sd SchemaDescription) error {
+	var visitErr error
+	obj.Visit(func(k []byte, v *fastjson.Value) {
+		fieldName := string(k)
+		fd, exists := sd.GetField(fieldName)
+		if !exists {
+			visitErr = NewErrFieldNotExist(fieldName)
+			return
+		}
+		if fd.IsPrimaryRelation() {
+			fd, exists = sd.GetField(fieldName + request.RelatedObjectID)
+			if !exists {
+				visitErr = NewErrFieldNotExist(fieldName)
+				return
+			}
+		}
+		val, err := validateFieldSchema(v, fd)
 		if err != nil {
-			return err
+			visitErr = err
+			return
 		}
-	}
-	return nil
-}
 
-// Set the value of a field.
-func (doc *Document) Set(field string, value any) error {
```

Comment:
question: It appears you have removed this function - the removal of this function does not make sense to me, why has it been removed?

Reply:
It was never used since only SetAs is relevant now. It might come back with your suggestion of adding the SchemaDescription to Document.

Reply:
✅